### PR TITLE
S05 us172 worldmap

### PR DESCRIPTION
**Pull request by Kileana27**

### CODE REVIEW

**Review by OCR-VH-DevWeb — APPROVED** [review]
Approved

File: src/app/app.routes.ts [hunk 1]
```diff
@@ -84,17 +83,23 @@ export const routes: Routes = [
...
     component: NewsCharacterComponent,
   },
-
   {
     path: 'creationPersonnage/secrets',
     component: SecretSelectionPageComponent,
   },
 
+  {
+    path: 'creationPersonnage/lieux',
```
Comment (Benja-san) [inline src/app/app.routes.ts:96]: use hyphen rather than camel cases for urls ?

File: src/app/components/norta-map/norta-map.component.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  constructor() {}
+
+  ngOnInit(): void {
+    this.apiDataService.getContinents().subscribe((data) => {
+      this.norta = data[1]
+      this.cities = this.norta.regions.flatMap((region:any) => region.cities)
+  })
+  }
+
+  selectedCity?: City= new City();
```
Comment (Benja-san) [inline src/app/components/norta-map/norta-map.component.ts:27]: remove the ? or remove the empty city object

File: src/app/components/world-map-mobile/world-map-mobile.component.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <h1>Carte du monde</h1>
+    <div class="world_map">
+        <img src="../assets/pictures/WorldMap.svg" alt="WorldMap" usemap="#worldmap">
+        <map name="worldmap">
+            <area shape="circle" coords="130,60,30" alt="norta" (click)="onClickNorta()" (mouseenter)="onMouseEnterNorta()" (mouseleave)="onMouseLeaveNorta()">
+            <area shape="circle" coords="200,65,20" alt="nelm" (click)="onClickNelm()" (mouseenter)="onMouseEnterNelm()" (mouseleave)="onMouseLeaveNelm()">
+        </map>
+    </div>
+    <div class="show_continent_description">
+        @if (norta == true) {
```
Comment (Benja-san) [inline src/app/components/world-map-mobile/world-map-mobile.component.html:12]: ===

File: src/app/components/world-map-mobile/world-map-mobile.component.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            <h2>Norta</h2>
+            <p>Le continent de Norta est partagé en Trois Royaumes, représentés par une capitale, où réside un.e Roi / Reine.</p>
+        }
+        @if (nelm == true) {
```
Comment (Benja-san) [inline src/app/components/world-map-mobile/world-map-mobile.component.html:16]: ===

File: src/app/components/world-map/world-map.component.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <h1>Carte du monde</h1>
+    <div class="world_map">
+        <img src="../assets/pictures/WorldMap.svg" alt="WorldMap" usemap="#worldmap">
+        <map name="worldmap">
+            <area shape="circle" coords="240,120,70" alt="norta" (click)="onClickNorta()" (mouseenter)="onMouseEnterNorta()" (mouseleave)="onMouseLeaveNorta()">
+            <area shape="circle" coords="380,120,30" alt="nelm" (click)="onClickNelm()" (mouseenter)="onMouseEnterNelm()" (mouseleave)="onMouseLeaveNelm()">
+        </map>
+    </div>
+    <div class="show_continent_description">
+        @if (norta == true) {
```
Comment (Benja-san) [inline src/app/components/world-map/world-map.component.html:12]: ===

File: src/app/components/world-map/world-map.component.html [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+            <h2>Norta</h2>
+            <p>Le continent de Norta est partagé en Trois Royaumes, représentés par une capitale, où réside un.e Roi / Reine.</p>
+        }
+        @if (nelm == true) {
```
Comment (Benja-san) [inline src/app/components/world-map/world-map.component.html:16]: ===

File: src/app/components/world-map-mobile/world-map-mobile.component.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    this.nelm = false;
+  }
+
+  onClickNelm(): void {
+    this.isClickedNelm = true;
+    this.nelm = true;
+    this.isClickedNorta = false;
+    this.norta = false;
+  }
+}
```
Comment (Benja-san) [inline src/app/components/world-map-mobile/world-map-mobile.component.ts:64]: all those methods can be refactored as one each (3 total) taking string as parameter

File: src/app/components/world-map/world-map.component.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  onClickNorta(): void {
+    this.isClickedNorta = true;
+    this.norta = true;
+    this.isClickedNelm = false;
+    this.nelm = false;
+    this.conditionNorta = true;
+    this.selectedContinent = 'norta';
+    console.log("envoie")
```
Comment (Benja-san) [inline src/app/components/world-map/world-map.component.ts:66]: remove console log

File: src/app/components/world-map/world-map.component.ts [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    this.sendNortaIsDisplayToParent.emit(true);
+  }
+
+  onClickNelm(): void {
+    this.isClickedNelm = true;
+    this.nelm = true;
+    this.isClickedNorta = false;
+    this.norta = false;
+    this.conditionNelm = true;
+  }
```
Comment (Benja-san) [inline src/app/components/world-map/world-map.component.ts:76]: same for those methods

File: src/app/pages/location-selection-page/location-selection-page.component.html [hunk 10]
```diff
@@ -0,0 +1,29 @@
+<div class="background_image">
+    <section class="location_container">
+        <div class="birth_place">
+            <h2>Lieu d'origine</h2>
```
Comment (Benja-san) [inline src/app/pages/location-selection-page/location-selection-page.component.html:4]: every page should have at least one h1

File: src/app/pages/location-selection-page/location-selection-page.component.ts [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  templateUrl: './location-selection-page.component.html',
+  styleUrl: './location-selection-page.component.scss'
+})
+export class LocationSelectionPageComponent {
+
+    isNortaDisplay: boolean | null = null;
+
+    onReceiveIsNortaDisplay(event: boolean): void {
+    this.isNortaDisplay=event;
+    console.log("reçu: " + this.isNortaDisplay);
```
Comment (Benja-san) [inline src/app/pages/location-selection-page/location-selection-page.component.ts:22]: remove clog

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**